src/main/webapp/features/shared/services/game/Game.service.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+angular.module('catan')
+        .factory('Game', ['GameModel', '$q', 'Remote', function (GameModel, $q, Remote) {
+
+            var Game = function(idOrDetails) {
```

Comment:
in the constructor you have main logic:
```javascript
                if (typeof idOrDetails === "number") {
                    id = idOrDetails;
                } else if (typeof idOrDetails === "object") {
                    id = idOrDetails[GAME_PRIMARY_KEY];
                    details = idOrDetails;
                    status = STATUS_LOADED;
                }
```

So if you create constructor taking 3 parameters: (Id, details, status), you can remove this logic at all and just call constructors in 2 different ways: Game(123) or Game(123, {object,with,details}, "StatusValue")